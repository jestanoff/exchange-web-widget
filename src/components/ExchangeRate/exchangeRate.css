.container {
  background: #ffffff;
  border-radius: 18px;
  border: 1px solid  #F0F0F0;
  color: #0b5ce4;
  margin-right: calc(50% - 100px);
  padding: 8px 16px;
}

.icon {
  fill: #0b5ce4;
  height: 16px;
  margin-right: 12px;
  vertical-align: bottom;
  width: 16px;
}

.rateSmaller {
  font-size: 13px;
}
