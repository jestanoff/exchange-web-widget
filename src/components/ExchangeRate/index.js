import React from 'react';
import { string } from 'prop-types';
import Icon from 'components/Icon';
import * as currencies from 'constants/currencies';
import styles from './exchangeRate.css';

const ExchangeRate = ({ from, rate, to }) => (
  <section className={styles.container}>
    <Icon name="IncliningArrow" className={styles.icon} />
    {`${currencies[from].symbol}1 = ${currencies[to].symbol}${rate}`}
  </section>
);

ExchangeRate.propTypes = {
  from: string.isRequired,
  rate: string,
  to: string.isRequired,
};

ExchangeRate.defaultProps = {
  rate: '',
};

export default ExchangeRate;
