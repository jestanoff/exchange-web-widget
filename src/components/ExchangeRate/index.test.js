import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ExchangeRate from './';


describe('ExchangeRate component', () => {
  test('should render proper markup', () => {
    const { container } = render(<ExchangeRate from="USD" rate="1.2" to="EUR" />);
  
    expect(container).toMatchSnapshot();
  });
});
