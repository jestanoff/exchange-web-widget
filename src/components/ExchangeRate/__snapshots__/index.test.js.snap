// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeRate component should render proper markup 1`] = `
<div>
  <section
    class="container"
  >
    <svg
      class="icon"
      focusable="false"
      viewBox="0 0 44 44"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m41.57605768882 11.31137948487h-7.33708622976c-1.35185813783 0-2.44565719593 1.09506011979-2.44565719593 2.44565719593s1.09391370007 2.44565719593 2.44565719593 2.44565719593h1.10835858858l-11.05836465192 10.26882538841-8.14909531984-6.11064640967c-.92905854384-.69862817944-2.23311097921-.64486109441-3.10255569744.13848750259l-12.22851526359 11.00562934464c-1.00311725797.90280553218-1.08428377439 2.44932573904-.18147824221 3.45358941674.9052130136 1.00197083825 2.44806467735 1.08554483609 3.45358941674.18147824221l10.73335466033-9.65973258937 8.17534833151 6.12979161905c.93868846952.71055094456 2.25707115143.64245361299 3.13121619052-.16359409453l13.00945637917-12.08028319336v1.72788380711c0 1.35059707614 1.09391370007 2.44565719593 2.44565719593 2.44565719593 1.35185813783 0 2.44565719593-1.09506011979 2.44565719593-2.44565719593v-7.33708622976c.00022928394-1.35059707614-1.09368441612-2.44565719593-2.44554255396-2.44565719593z"
      />
    </svg>
    $1 = â‚¬1.2
  </section>
</div>
`;
